### PR TITLE
Update CI script logic for yarn custom shell

### PR DESCRIPTION
**Pull request by LeonFLK**
## fixes KILTProtocol/ticket#854

yarn 2 berry uses a new custom shell to provide consistent behaviour across systems. This breaks the test:integration script, that checks whether local chain is running on port 9944.
if -> then -> is written {condition && command} || exit

Also updates to latest berry version of 2.4.0

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [x] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

**Comment by LeonFLK** [comment]
Not sure if updating to 2.4.0 is neccessary.

**Comment by tjwelde** [comment]
updating yarn to 2.4.0 is not necessary, but also doesn't hurt, so let's keep it

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "test": "jest --coverage --group=-integration",
     "test:ci": "jest --ci --coverage --group=-integration",
     "test:integration:run": "jest --group=integration -b -w 3 --silent",
-    "test:integration": "if lsof -i :9944 > /dev/null; then yarn test:integration:run; else echo 'Can not connect to chain. Is it running?'; exit 1;fi",
+    "test:integration": "{lsof -i :9944 > /dev/null && yarn test:integration:run;} || echo 'Can not connect to chain. Is it running?'; exit 1;fi",
```
Comment (tjwelde) [inline package.json:11]: since you got rid of `if`, you can also remove `fi`

Comment (tjwelde) [reply package.json:11]: Also, can you add a space after `{` and before `}`? It would make it more standard conform

**Review by tjwelde — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (LeonFLK) [reply package.json:11]: whoopsie

Comment (LeonFLK) [reply package.json:11]: and yes ofc

**Pull request merged**